src/components/dialog/dialog.js

```diff
@@ -114,6 +117,20 @@ const VUIDialog = ({
         dialog.setAttribute('tabindex', 1);
         dialog.setAttribute('aria-hidden', false);
 
+        // Alerts and modal dialog types don't have close buttons
+        if (hasCloseBtn && !isAlert && !isModal) {
+            const closeBtnEl = createEl({
+                element: 'button',
+                className: ['js-dialog-btn-close', closeButtonClassName],
```

Comment:
It happens when the dialog is launched, it checks for all buttons with that js-hook and attaches an event listener to close it. Link ^^